client/src/js/components/bhDateInterval.js

```diff
@@ -48,6 +48,8 @@ function bhDateInterval(bhConstants, Fiscal, Session, PeriodService, $translate)
   // expose to the view
   $ctrl.search = search;
   $ctrl.clear = clear;
+  $ctrl.lastDateFrom = null;
```

Comment:
Do you need to attach these variables to `$ctrl`?  I assume you could just keep them scoped to this component with `let lastDateFrom = null;` and so on, since you don't use them in the template.

This isn't a big deal, just an observation.